modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_0/UpgradeDLFileEntryType.java

```diff
@@ -14,9 +14,221 @@
 
 package com.liferay.document.library.internal.upgrade.v3_2_0;
 
+import com.liferay.document.library.kernel.model.DLFileEntryMetadata;
+import com.liferay.document.library.kernel.model.DLFileEntryType;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.kernel.service.ResourceLocalServiceUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.uuid.PortalUUIDUtil;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
 /**
  * @author Adolfo Pérez
  */
 public class UpgradeDLFileEntryType
 	extends com.liferay.portal.upgrade.v7_3_x.UpgradeDLFileEntryType {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		super.doUpgrade();
+
+		try (PreparedStatement ps1 = connection.prepareStatement(
+				StringBundler.concat(
+					"select companyId, fileEntryTypeId, fileEntryTypeKey, ",
+					"groupId, name, userId, userName from DLFileEntryType ",
+					"where (dataDefinitionId IS NULL OR dataDefinitionId = 0) ",
+					"and fileEntryTypeKey != 'BASIC-DOCUMENT'"));
+			PreparedStatement ps2 = AutoBatchPreparedStatementUtil.autoBatch(
+				connection.prepareStatement(
+					"update DLFileEntryType set dataDefinitionId = ? where " +
+						"fileEntryTypeId = ? "));
+			ResultSet rs = ps1.executeQuery()) {
+
+			String ddmStructureDefinition = StringUtil.read(
```

Comment:
This value (and the layout def as well) doesn't change, so there's no need to pass it as an argument in each call. Why not save them in instance variables, initialized in the constructor?

Reply:
Doing this forces us to do an ugly try-catch inside the constructor. I think it is not worth it.